cleaning tool errors

Danilodum opened the pull request:
preparing for merge in master

Commit:
fixed checkstyle and PMD errors

Commit:
moved CurveState under right package

jonathanraes commented on src/main/java/com/funkydonkies/w4v3/curve/CustomCurveMesh.java:
```diff
@@ -67,17 +67,19 @@ public void setMeshPoints(final int i) {
-			meshStructurePoints[splinePoints.length * 3 + 2 * i] = splinePoints[i];
-			meshStructurePoints[splinePoints.length * 3 + 2 * i + 1] = vecNew;		
+			meshStructurePoints[splinePoints.length * mulLength + offset * i] = splinePoints[i];
+			meshStructurePoints[splinePoints.length * mulLength + offset * i + 1] = vecNew;		
```
looks very clean, only thing maybe add parenthesis on these kind of things to make it even more clear

Danilodum replied:
I agree

Commit:
fixed some bugs in isBgSet() and more

Commit:
fixed a real bug in CurveState

what is left of findbugs errors I don't know how to fix, we need this exposed...

Commit:
added braces in expr in CustomCurveMesh

jonathanraes commented:
nice looks clean

jonathanraes commented:
Still some findbugs errors regarding splinepoints and curvepoints though. We're still looking for a solution on that.
Can be merged into customcurvemesh
